blob/blob.go

```diff
@@ -833,65 +831,74 @@ type WriterOptions struct {
 	BeforeWrite func(asFunc func(interface{}) bool) error
 }
 
-// FromURLFunc is intended for use by provider implementations.
-// It allows providers to convert a parsed URL from Open to a driver.Bucket.
-type FromURLFunc func(context.Context, *url.URL) (driver.Bucket, error)
-
-var (
-	// registry maps scheme strings to provider-specific instantiation functions.
-	registry = map[string]FromURLFunc{}
-	// registryMu protected registry.
-	registryMu sync.Mutex
-)
+// A type that implements BucketURLOpener can open buckets based on a URL.
+// The opener must not modify the URL argument. OpenBucketURL must be safe to
+// call from multiple goroutines.
+type BucketURLOpener interface {
```

Comment:
Add back "It is intended for use by provider implementations."?

I think we should try to make it clear to `blob.Bucket` users which parts of the package they can/should ignore.